backend/Utils/Edna.Utils.Http/Edna.Utils.Http/HttpClaimsExtension.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+
+namespace Edna.Utils.Http
+{
+    public static class HttpClaimsExtension
+    {
+        private static readonly string[] PossibleEmailClaimTypes = { "email", "upn", "unique_name" };
+        public static bool TryGetUserEmails(this IHeaderDictionary headers, out List<string> userEmails, Action<string> logAction = null)
+        {
+            userEmails = new List<string>();
+
+            if (!headers.TryGetTokenClaims(out Claim[] claims, logAction))
+            {
+                logAction("Error in sent JWT");
+                return false;
+            }
+
+            // By checking appidacr claim, we can know if the call was made by a user or by the system.
+            // https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens
+            string appidacr = claims.FirstOrDefault(claim => claim.Type == "appidacr")?.Value;
```

Comment:
I was thinking about refactoring this into well-named method to indicate the overall intent here.